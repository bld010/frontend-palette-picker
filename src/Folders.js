import React from "react";
import { FaTrash } from "react-icons/fa";
import "./Folders.scss";

export const Folders = props => {

  const handleDisplayFolderEnter = (e, id) => {
    if (e.keyCode === 13) {
      props.displayFolderPalettes(id);
    }
  };

  const handleDeleteFolderEnter = (e, folder) => {
    if (e.keyCode === 13){
      props.deleteFolder(folder)
    }
  }

  let alphabetizedFolders = props.folders.sort((a,b) => {
    var folderNameA = a.name.toUpperCase();
    var folderNameB = b.name.toUpperCase();

    if (folderNameA < folderNameB) {
      return -1
    } else {
      return 1
    }
  })

  let foldersList = alphabetizedFolders.map(folder => {
    return (
      <li key={folder.id}>
        <p tabIndex={0} onKeyDown={e => handleDisplayFolderEnter(e, folder.id)} onClick={() => props.displayFolderPalettes(folder.id)}>
          {folder.name}</p>
        <FaTrash tabIndex={0} onKeyDown={e=> handleDeleteFolderEnter(e, folder)} onClick={() => props.deleteFolder(folder)}/>
      </li>
    );
  });

  return (
    <section className="Folders">
      <ul>{foldersList}</ul>
    </section>
  );
};

