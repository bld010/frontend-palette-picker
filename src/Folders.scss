.Folders {
  // border: 1px solid black;
  // width: 300px;
  // height: 300px;
  
  & ul {
    overflow: scroll;
    width: 100%;
    height: 28vh;
    min-height: 200px;
    border: 1px dashed lightgrey;

    // box-shadow: inset 0px 0px 5px 5px rgba(0, 0, 0, 0.2);
    padding-top: 20px;

    & li {
      border-bottom: 0.5px dotted rgb(202, 202, 202);
      padding-bottom: 10px;
      font-size: 18px;
      list-style-type: none;
      margin: 12px 0 12px 0;
      cursor: pointer;
      width: 90%;
      font-family: "Open Sans", sans-serif;
      display: inline-flex;
      justify-content: space-between;
      align-items: center;

      & .trash {
        opacity: 0.3;
      }
    }


    & li:focus {
      background-color: rgb(191, 230, 243);
      outline: none;
    }
  }
}