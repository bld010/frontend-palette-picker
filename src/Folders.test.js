import React from "react";
import { shallow } from "enzyme";
import {Folders} from "./Folders";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("Folders", () => {

  let mockDisplayFolderPalettes = jest.fn();
  let mockFolders = [
    {
      id: 84,
      name: "Almost Blues",
      created_at: "2019-10-15T01:01:13.675Z",
      updated_at: "2019-10-15T01:01:13.675Z"
    },
    {
      id: 85,
      name: "Fuzzy Purple",
      created_at: "2019-10-15T01:01:50.370Z",
      updated_at: "2019-10-15T01:01:50.370Z"
    },
    {
      id: 86,
      name: "Rubber Ducky Takes a Bath",
      created_at: "2019-10-15T01:01:13.675Z",
      updated_at: "2019-10-15T01:01:13.675Z"
    },
    {
      id: 87,
      name: "Money Greens",
      created_at: "2019-10-15T01:01:50.370Z",
      updated_at: "2019-10-15T01:01:50.370Z"
    }
  ];

  let mockDeleteFolder = jest.fn();

  let wrapper = shallow(
    <Folders
      displayFolderPalettes={mockDisplayFolderPalettes}
      folders={mockFolders}
      deleteFolder={mockDeleteFolder}
    />
  );

  it("should match the snapshots", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleDeleteFolderEnter on the keydown of a trash', () => {
    // wrapper.instance().handleDeleteFolderEnter = jest.fn()
    wrapper.find('.trash').first().simulate('keydown', {keyCode: 13})
    expect(wrapper.instance().deleteFolderPalettes).toHaveBeenCalled()
  });

  it('should call handleDisplayFolderEnter on the keydown of a folder', () => {
    wrapper.find('p').first().simulate('keydown', {keyCode: 13})
    expect(wrapper.instance().handleDisplayFolderEnter).toHaveBeenCalled()
  });

  it('should call deleteFolder on the click of a trash', () => {
    wrapper.find('.faTrash').first().simulate('click')
    expect(mockDeleteFolder).toHaveBeenCalled()
  });

  it('should call displayFolderPalettes on the click of a folder', () => {
    wrapper.find('p').first().simulate('click')
    expect(mockDisplayFolderPalettes).toHaveBeenCalled()
  });

});
