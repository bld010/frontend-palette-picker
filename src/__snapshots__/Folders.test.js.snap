// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Folders should match the snapshots 1`] = `
<section
  className="Folders"
>
  <ul>
    <li
      key="84"
    >
      <p
        className="folder-list"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
      >
        Almost Blues
      </p>
      <div
        className="trash"
      >
        <FaTrash
          className="faTrash"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        />
      </div>
    </li>
    <li
      key="85"
    >
      <p
        className="folder-list"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
      >
        Fuzzy Purple
      </p>
      <div
        className="trash"
      >
        <FaTrash
          className="faTrash"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        />
      </div>
    </li>
    <li
      key="87"
    >
      <p
        className="folder-list"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
      >
        Money Greens
      </p>
      <div
        className="trash"
      >
        <FaTrash
          className="faTrash"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        />
      </div>
    </li>
    <li
      key="86"
    >
      <p
        className="folder-list"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
      >
        Rubber Ducky Takes a Bath
      </p>
      <div
        className="trash"
      >
        <FaTrash
          className="faTrash"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        />
      </div>
    </li>
  </ul>
</section>
`;
